src/Microsoft.ML.Transforms/TermLookupTransform.cs

```diff
@@ -514,11 +546,21 @@ private static byte[] GetBytesFromDataView(IHost host, IDataView lookup, string
                 if (!isBinary && !isTranspose)
                     throw host.ExceptUserArg(nameof(Arguments.Loader), "must specify the loader");
                 host.Assert(isBinary != isTranspose); // One or the other must be true.
-                sub = new SubComponent<IDataLoader, SignatureDataLoader>(isBinary ? "BinaryLoader" : "TransposeLoader");
+                if (isBinary)
+                {
+                    loader = new BinaryLoader(host, new BinaryLoader.Arguments(), fileSource);
+                }
+                else
+                {
+                    loader = new TransposeLoader(host, new TransposeLoader.Arguments(), fileSource);
+                }
```

Comment:
loader = isBinary? new BinaryLoader(host, new BinaryLoader.Arguments(), fileSource) : new TransposeLoader(host, new TransposeLoader.Arguments(), fileSource)

That is generally the pattern in the code.

Reply:
Unfortunately that code doesn't compile, and I'm not sure why.

```
Severity Code Description Project File Line Suppression State
Error CS0173 Type of conditional expression cannot be determined because there is no implicit conversion between 'Microsoft.ML.Runtime.Data.IO.BinaryLoader' and 'Microsoft.ML.Runtime.Data.IO.TransposeLoader' Microsoft.ML.Transforms F:\git\machinelearning\src\Microsoft.ML.Transforms\TermLookupTransform.cs 549 Active
```

---
In reply to: [212153383](https://github.com/dotnet/machinelearning/pull/700#discussion_r212153383) [](ancestors = 212153383)